tests/e2e/failover_test.go

```diff
@@ -24,6 +24,7 @@ import (
 	"github.com/stretchr/testify/require"
 	"google.golang.org/grpc"
 	_ "google.golang.org/grpc/health"
+	_ "google.golang.org/grpc/xds"
```

Comment:
This is used to import https://github.com/grpc/grpc-go/blob/master/xds/internal/balancer/outlierdetection/balancer.go#L57-L61

Reply:
that's really awesome @chaochn47! I'll give this a bash on Friday when I find some time.

Reply:
works pretty well, I just gave this a quick run with an apiserver. 
We also want to tune the retry policies in the future, I'm still a little unsure how to expose this in the etcd client. Exposing the raw service config is probably not a good idea given the potential changes to xds...

Reply:
> works pretty well, I just gave this a quick run with an apiserver.

Thanks! That's a good data point.

> We also want to tune the retry policies in the future, I'm still a little unsure how to expose this in the etcd client. 

Retry policy is on my to-do list to explore. Will ping you when I find a reliable way to do this with etcd client.

> Exposing the raw service config is probably not a good idea given the potential changes to xds...

Changes to xds should not affect how we configure outlier detection load balancing policy. It is part of the service config proto definition in the main grpc repo. https://github.com/grpc/grpc-proto/blob/master/grpc/service_config/service_config.proto#L241